Sources/Datadog/Logging/Log/LogBuilder.swift

```diff
@@ -22,10 +22,13 @@ internal struct LogBuilder {
     let networkConnectionInfoProvider: NetworkConnectionInfoProviderType?
     /// Shared mobile carrier info provider (or `nil` if disabled for given logger).
     let carrierInfoProvider: CarrierInfoProviderType?
+    /// Adjusts log's time (device time) to server time.
+    let dateCorrection: NTPDateCorrectionType?
 
     func createLogWith(level: LogLevel, message: String, date: Date, attributes: LogAttributes, tags: Set<String>) -> Log {
+        let dateInServerTime = dateCorrection?.toServerDate(deviceDate: date)
```

Comment:
This is optional because of the internal logger (used when `Datadog.debug` is set), which doesn't need time correction - it prints logs with device time.

For comparison, it's non-optional in `SpanBuilder` as we don't have any `Tracer` use case where we don't want the time correction to be applied.